app/controllers/verify/finance_controller.rb

```diff
@@ -17,8 +18,25 @@ def create
 
     private
 
+    def step
+      @_step ||= Idv::FinancialsStep.new(
+        idv_form: idv_finance_form,
+        idv_session: idv_session,
+        analytics: analytics,
+        params: step_params
+      )
+    end
+
+    def step_params
+      params.require(:idv_finance_form).permit(:finance_type, *Idv::FinanceForm::FINANCE_TYPES)
+    end
+
+    def confirm_step_needed
+      redirect_to verify_phone_path if idv_session.financials_confirmation.try(:success?)
+    end
```

Comment:
Understood. What I was implying would be saved in the DB is only a boolean to determine whether or not the user had completed a particular step. This would allow that step completion state to be remember past the session.